### PR TITLE
[documentation] Include instructions for setting up a local dev environment

### PR DESCRIPTION
**Pull request by obitech**
**What this PR does / why we need it**:

There are no instructions how to setup a local development environment. Those basic steps are somewhat copied in part from [Kubernetes' Github workflow](https://github.com/kubernetes/community/blob/master/contributors/guide/github-workflow.md) which should be enough to get people started with hacking on the project.

**Special notes for your reviewer**:
None.

**Does this PR introduce a user-facing and/or backwards incompatible change?**:

```release-note
NONE
```

**Does this PR require updating code package or user-facing documentation?**:

```documentation-note
Updates developer documentation in DEVELOPER.md
```

### CODE REVIEW

File: DEVELOPER.md [hunk 1]
```diff
@@ -1,5 +1,48 @@
...
+# Set this to your Github user
+export user={your github profile name}
+
+# Clone your fork
+cd $working_dir
+git clone git@github.com:$user/m3.git
+# or: https://github.com/$user/m3.git
+
+# Set upstream
+git remote add upstream git@github.com:m3db/m3.git
```
Comment (robskillington) [inline DEVELOPER.md]: Perhaps `cd` into `${working_dir}/m3` first?

File: DEVELOPER.md [hunk 2]
```diff
@@ -1,5 +1,48 @@
...
 
+## Setup your development environment
+
+First fork https://github.com/m3db/m3 into your own workspace on Github.
+
+Then create your clone:
+
+```bash
+export working_dir=$GOPATH/src/github.com/m3db
+mkdir -p $workding_dir
```
Comment (robskillington) [inline DEVELOPER.md]: nit: spelling here `working_dir` I believe.

File: DEVELOPER.md [hunk 3]
```diff
@@ -1,5 +1,48 @@
...
+
+# Set this to your Github user
+export user={your github profile name}
```
Comment (robskillington) [inline DEVELOPER.md]: super nit: `export user="your_github_profile_name"` just so for those not super familiar with bash they know which quotes to use?

File: DEVELOPER.md (hunk 1 above)
Comment (obitech) [reply DEVELOPER.md]: See line 17 :)

File: DEVELOPER.md (hunk 3 above)
Comment (obitech) [reply DEVELOPER.md]: Agreed, this might be better

File: DEVELOPER.md (hunk 2 above)
Comment (obitech) [reply DEVELOPER.md]: True, will fix

File: DEVELOPER.md (hunk 1 above)
Comment (obitech) [reply DEVELOPER.md]: You're right of coure

File: DEVELOPER.md (hunk 2 above)
Comment (obitech) [reply DEVELOPER.md]: Fixed

File: DEVELOPER.md (hunk 3 above)
Comment (obitech) [reply DEVELOPER.md]: Fixed

**Pull request merged**